app/src/main/java/com/infomaniak/drive/extensions/PairExtensions.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Infomaniak kDrive - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.drive.extensions
+
+fun <A : Any?, B : Any?> Pair<A?, B?>.letAll(callback: (value1: A, value2: B) -> Unit) {
+    if (first != null && second != null) callback.invoke(first!!, second!!)
+}
```

Comment:
I don't see the point, and it's not easy to understand unless you come and read the code.
And even if you wanted something like that, you should have used an easier-to-understand name like `bothNotNull` with the keyword `infix` to make it an operation.
Here, this function is only used once, so what's the reason? Why not just use a simple `if`?

Reply:
This is for people who want to use that extension to avoid checking the first and second value of a Pair. I think Isaw some of that check but I didn't want to change everywhere we did that check.

I named it `letAll` to make it a bit like `let` to tell `this extension is to check the nullability of both values` of the Pair.

Anyway, if you don't like it. I'll remove it.

Reply:
Yes, remove it, when we need it we'll do it but differently